components/net/fetch/request.rs

```diff
@@ -71,6 +84,7 @@ pub struct Request {
     pub preserve_content_codings: bool,
     // pub client: GlobalRef, // XXXManishearth copy over only the relevant fields of the global scope,
                               // not the entire scope to avoid the libscript dependency
+    pub is_service_worker_global_scope: Option<bool>,
```

Comment:
oh, I see.

I'll look through this sometime tomorrow I guess (got some changes to review in humpty_dumpty first)